WAR containerization broken with the base image change

Glad we had a tomcat integration test and were able to catch this before making it out.

In #3124, we were computing the entrypoint while mistakenly assuming that the base image is always the default Jetty image.

```
com.google.cloud.tools.jib.gradle.WarProjectIntegrationTest > testBuild_tomcatServlet25 FAILED
    java.lang.AssertionError: expected:<Hello world> but was:<null>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at com.google.cloud.tools.jib.gradle.WarProjectIntegrationTest.verifyBuildAndRun(WarProjectIntegrationTest.java:63)
        at com.google.cloud.tools.jib.gradle.WarProjectIntegrationTest.testBuild_tomcatServlet25(WarProjectIntegrationTest.java:52)

    java.lang.RuntimeException: Command 'docker stop ac3948ea573d7b08e5e320ad02a5772aeb5d1f58491cee99a340d914860da171' failed: Error response from daemon: No such container: ac3948ea573d7b08e5e320ad02a5772aeb5d1f58491cee99a340d914860da171
        at com.google.cloud.tools.jib.Command.run(Command.java:79)
        at com.google.cloud.tools.jib.Command.run(Command.java:52)
        at com.google.cloud.tools.jib.gradle.WarProjectIntegrationTest.tearDown(WarProjectIntegrationTest.java:40)
```

--- a/jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java
+++ b/jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java
@@ -543,7 +543,8 @@ public class PluginConfigurationProcessor {
    * <ol>
    *   <li>null (inheriting from the base image), if the user specified value is {@code INHERIT}
    *   <li>the user specified one, if set
-   *   <li>for a WAR project, {@code ["java", "-jar", "/usr/local/jetty/start.jar"]}
+   *   <li>for a WAR project, null (inheriting) if a custom base image is specified, and {@code
+   *       ["java", "-jar", "/usr/local/jetty/start.jar"]} otherwise (default Jetty base image)
    *   <li>for a non-WAR project, by resolving the main class
    * </ol>
    *
@@ -589,7 +590,9 @@ public class PluginConfigurationProcessor {
                 "mainClass, extraClasspath, jvmFlags, and expandClasspathDependencies "
                     + "are ignored for WAR projects"));
       }
-      return Arrays.asList("java", "-jar", "/usr/local/jetty/start.jar");
+      return rawConfiguration.getFromImage().isPresent()
+          ? null
+          : Arrays.asList("java", "-jar", "/usr/local/jetty/start.jar");
     }
 
     List<String> classpath = new ArrayList<>(rawExtraClasspath);

--- a/jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java
+++ b/jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java
@@ -569,6 +569,22 @@ public class PluginConfigurationProcessorTest {
   }
 
   @Test
+  public void testWebAppEntrypoint_inheritedFromCustomBaseImage()
+      throws InvalidImageReferenceException, IOException, CacheDirectoryCreationException,
+          MainClassInferenceException, InvalidAppRootException, InvalidWorkingDirectoryException,
+          InvalidPlatformException, InvalidContainerVolumeException,
+          IncompatibleBaseImageJavaVersionException, NumberFormatException,
+          InvalidContainerizingModeException, InvalidFilesModificationTimeException,
+          InvalidCreationTimeException {
+    when(projectProperties.isWarProject()).thenReturn(true);
+    when(rawConfiguration.getFromImage()).thenReturn(Optional.of("custom-base-image"));
+
+    BuildContext buildContext = getBuildContext(processCommonConfiguration());
+
+    assertThat(buildContext.getContainerConfiguration().getEntrypoint()).isNull();
+  }
+
+  @Test
   public void testGetAppRootChecked() throws InvalidAppRootException {
     when(rawConfiguration.getAppRoot()).thenReturn("/some/root");
 

